db/model/ExplorerViewsHelpers.ts

```diff
@@ -0,0 +1,279 @@
+import * as db from "../db.js"
+import {
+    ExplorerChartCreationMode,
+    ExplorerProgram,
+    ExplorerGrapherInterface,
+} from "@ourworldindata/explorer"
+import {
+    CoreRow,
+    DimensionProperty,
+    GrapherInterface,
+    OwidChartDimensionInterface,
+    OwidColumnDef,
+} from "@ourworldindata/types"
+import {
+    excludeUndefined,
+    mergeGrapherConfigs,
+    omitUndefinedValues,
+    parseIntOrUndefined,
+    PartialBy,
+    RequiredBy,
+} from "@ourworldindata/utils"
+import { defaultGrapherConfig } from "@ourworldindata/grapher"
+import { getChartConfigById } from "./Chart.js"
+import { getMergedGrapherConfigForVariable } from "./Variable.js"
+import * as R from "remeda"
+
+function columnDefToDimensionObject(
+    colDef?: OwidColumnDef
+): Partial<OwidChartDimensionInterface> {
+    if (!colDef) return {}
+    return omitUndefinedValues({
+        variableId: colDef.owidVariableId,
+        slug: colDef.slug,
+        // pipe all fields of the explorer's column grammar into the display field
+        // this is not necessarilly correct; we need to add a translation step here
```

Comment:
@danyx23, my implementation wasn't complete and this is one of the more major pieces missing. `OwidColumnDef` doesn't correspond to the `display` field one to one, so we need a translation step here.

The other big piece I also haven’t gotten around to is adding support for the `duplicate` transform column.

More generally, my implementation was done before the Grapher state refactor, and I ended up abandoning it because I was hoping there’d be a way to use GrapherState that would save us from having to rebuild the configs from scratch like I did here. Do you think that's worth exploring?